metrics/serve_test.go

```diff
@@ -158,6 +161,48 @@ func TestRunMetricsWithInvalidSeriesCounts(t *testing.T) {
 
 	promRegistry = prometheus.NewRegistry()
 
-	_, err := RunMetrics(metricCount, labelCount, seriesCount, seriesChangeRate, maxSeriesCount, minSeriesCount, metricLength, labelLength, valueInterval, seriesInterval, metricInterval, seriesChangeInterval, operationMode, []string{constLabel}, stop)
+	_, err := RunMetrics(metricCount, labelCount, seriesCount, seriesChangeRate, maxSeriesCount, minSeriesCount, metricLength, labelLength, valueInterval, seriesInterval, metricInterval, seriesChangeInterval, spikeMultiplier, operationMode, []string{constLabel}, stop)
 	assert.Error(t, err)
 }
+
+func TestRunMetricsSpikeChange(t *testing.T) {
+	const (
+		metricCount          = 1
+		labelCount           = 1
+		initialSeriesCount   = 100
+		maxSeriesCount       = 30
+		minSeriesCount       = 10
+		spikeMultiplier      = 1.5
+		seriesChangeRate     = 10
+		metricLength         = 1
+		labelLength          = 1
+		valueInterval        = 100
+		seriesInterval       = 100
+		metricInterval       = 100
+		seriesChangeInterval = 10
+		operationMode        = "spike"
+		constLabel           = "constLabel=test"
+	)
+
+	stop := make(chan struct{})
+	defer close(stop)
+
+	promRegistry = prometheus.NewRegistry()
+
+	_, err := RunMetrics(metricCount, labelCount, initialSeriesCount, seriesChangeRate, maxSeriesCount, minSeriesCount, metricLength, labelLength, valueInterval, seriesInterval, metricInterval, seriesChangeInterval, spikeMultiplier, operationMode, []string{constLabel}, stop)
+	assert.NoError(t, err)
+
+	time.Sleep(2 * time.Second)
+	for i := 0; i < 4; i++ {
+		time.Sleep(time.Duration(seriesChangeInterval) * time.Second)
+		if i%2 == 0 {
+			currentCount := countSeries(t, promRegistry)
+			expectedCount := initialSeriesCount
+			assert.Equal(t, expectedCount, currentCount, "Halved series count should be %d but got %d", int(expectedCount), currentCount)
+		} else {
+			currentCount := countSeries(t, promRegistry)
+			expectedCount := initialSeriesCount * spikeMultiplier
+			assert.Equal(t, int(expectedCount), currentCount, "Doubled series count should be %d but got %d", int(expectedCount), float64(currentCount))
```

Comment:
we should change the messages here, you could put `"multiplied the series count by %d, should be %d but got %d"` for example

Reply:
changed